### PR TITLE
[Admin][Product] Add behat tests for filtering and check/uncheck all taxons

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | N/A
| License         | MIT

### CODE REVIEW

**Review by mpysiak — DISMISSED** [review]

File: features/admin/product/managing_products/checking_taxons_of_product.feature [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        Given the store operates on a single channel in "United States"
+        And the store classifies its products as "Clothes" and "T-Shirts"
+        And the store has a "Shirt" configurable product
+        And the store has a "T-Shirt" configurable product
+        And the product "T-Shirt" belongs to taxon "Clothes"
+        And the product "T-Shirt" belongs to taxon "T-Shirts"
+        And I am logged in as an administrator
+
+    @ui @no-api @mink:chromedriver
+    Scenario: Check all taxons
```
Comment (GSadee) [inline features/admin/product/managing_products/checking_taxons_of_product.feature]:
```suggestion
    Scenario: Checking all taxons
```

File: features/admin/product/managing_products/checking_taxons_of_product.feature [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        When I want to modify the "Shirt" product
+        And I check all taxons
+        And I save my changes
+        Then the product "Shirt" should have the "Clothes" taxon
+        And the product "Shirt" should have the "T-Shirts" taxon
+
+    @ui @no-api @mink:chromedriver
+    Scenario: Uncheck all taxons
```
Comment (GSadee) [inline features/admin/product/managing_products/checking_taxons_of_product.feature]:
```suggestion
    Scenario: Unchecking all taxons
```

File: features/admin/product/managing_products/checking_taxons_of_product.feature [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        When I want to modify the "T-Shirt" product
+        And I uncheck all taxons
+        And I save my changes
+        Then the product "Shirt" should not have the "Clothes" taxon
+        And the product "Shirt" should not have the "T-Shirts" taxon
+
+    @ui @no-api @mink:chromedriver
+    Scenario: Check all filtered taxons
```
Comment (GSadee) [inline features/admin/product/managing_products/checking_taxons_of_product.feature]:
```suggestion
    Scenario: Checking all filtered taxons
```

File: features/admin/product/managing_products/checking_taxons_of_product.feature [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        When I want to modify the "Shirt" product
+        And I filter taxons by "T-Shirts"
+        And I check all taxons
+        And I save my changes
+        Then the product "Shirt" should have the "T-Shirts" taxon
+        And the product "Shirt" should not have the "Clothes" taxon
+
+    @ui @no-api @mink:chromedriver
+    Scenario: Uncheck all filtered taxons
```
Comment (GSadee) [inline features/admin/product/managing_products/checking_taxons_of_product.feature]:
```suggestion
    Scenario: Unchecking all filtered taxons
```

File: features/admin/product/managing_products/checking_taxons_of_product.feature [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        And the product "Shirt" should not have the "T-Shirts" taxon
+
+    @ui @no-api @mink:chromedriver
+    Scenario: Check all filtered taxons
+        When I want to modify the "Shirt" product
+        And I filter taxons by "T-Shirts"
+        And I check all taxons
+        And I save my changes
+        Then the product "Shirt" should have the "T-Shirts" taxon
+        And the product "Shirt" should not have the "Clothes" taxon
```
Comment (GSadee) [inline features/admin/product/managing_products/checking_taxons_of_product.feature]:
```suggestion
        But the product "Shirt" should not have the "Clothes" taxon
```

File: features/admin/product/managing_products/filtering_taxons_of_product.feature [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        And the product "T-Shirt" belongs to taxon "Clothes"
+        And the product "T-Shirt" belongs to taxon "T-Shirts"
+        And I am logged in as an administrator
+
+    @ui @no-api @mink:chromedriver
+    Scenario: Filter product taxons
+        When I want to modify the "Shirt" product
+        And I filter taxons by "T-Shirts"
+        Then I should see the "T-Shirts" taxon
+        And I should not see the "Clothes" taxon
```
Comment (GSadee) [inline features/admin/product/managing_products/filtering_taxons_of_product.feature]:
```suggestion
        But I should not see the "Clothes" taxon
```

File: features/admin/product/managing_products/filtering_taxons_of_product.feature [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        Given the store operates on a single channel in "United States"
+        And the store classifies its products as "Clothes" and "T-Shirts"
+        And the store has a "Shirt" configurable product
+        And the store has a "T-Shirt" configurable product
+        And the product "T-Shirt" belongs to taxon "Clothes"
+        And the product "T-Shirt" belongs to taxon "T-Shirts"
+        And I am logged in as an administrator
+
+    @ui @no-api @mink:chromedriver
+    Scenario: Filter product taxons
```
Comment (GSadee) [inline features/admin/product/managing_products/filtering_taxons_of_product.feature]:
```suggestion
    Scenario: Filtering product taxons
```

File: features/admin/product/managing_products/filtering_taxons_of_product.feature [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    As an Administrator
+    I want to search for a specific taxon
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store classifies its products as "Clothes" and "T-Shirts"
+        And the store has a "Shirt" configurable product
+        And the store has a "T-Shirt" configurable product
+        And the product "T-Shirt" belongs to taxon "Clothes"
+        And the product "T-Shirt" belongs to taxon "T-Shirts"
```
Comment (GSadee) [inline features/admin/product/managing_products/filtering_taxons_of_product.feature]: It does not seem to be needed here 🤔 
```suggestion
```

File: src/Sylius/Behat/Context/Ui/Admin/ManagingProductTaxonsContext.php [hunk 9]
```diff
@@ -44,6 +44,27 @@ public function iChangeThatTheProductDoesNotBelongToTheTaxon(
...
+    public function iUncheckAllTaxons(): void {
+        $this->taxonomyFormElement->uncheckAllTaxons();
+    }
+
+    /**
+     * @When I filter taxons by :phrase
+     */
+    public function iFilterTaxonsBy(string $phrase): void {
+        $this->taxonomyFormElement->filterTaxonsBy($phrase);
+    }
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingProductTaxonsContext.php]:
```suggestion
    /**
     * @When I check all taxons
     */
    public function iCheckAllTaxons(): void 
    {
        $this->taxonomyFormElement->checkAllTaxons();
    }

    /**
     * @When I uncheck all taxons
     */
    public function iUncheckAllTaxons(): void 
    {
        $this->taxonomyFormElement->uncheckAllTaxons();
    }

    /**
     * @When I filter taxons by :phrase
     */
    public function iFilterTaxonsBy(string $phrase): void 
    {
        $this->taxonomyFormElement->filterTaxonsBy($phrase);
    }
```

**Review by GSadee — APPROVED** [review]

**Pull request merged**